wpt-sync/sync.js

```diff
@@ -142,20 +144,29 @@ async function updatePullRequests(dir, localBranches, remote, remoteBranches) {
             const affectedFiles = git(`diff --name-only ${branch} ${branch}^`).replace(/\n/g, ' ');
             const diff = git(`diff ${branch} ${remote}/${branch} -- ${affectedFiles}`).trim();
             if (diff) {
-                git(`push -f ${remote} ${branch}`);
-                console.log(`  Updated remote branch: ${branch}`);
+                if (dryRun) {
+                    console.log(`  Would updated remote branch: ${branch} (skipped in dry run)`);
+                } else {
+                    git(`push -f ${remote} ${branch}`);
+                    console.log(`  Updated remote branch: ${branch}`);
+                }
             } else {
                 console.log(`  Remote branch is up to date: ${branch}`);
             }
         } else {
-            git(`push ${remote} ${branch}`);
-            console.log(`  Created new remote branch: ${branch}`);
+            if (dryRun) {
+                console.log(`  Would create new remote branch: ${branch} (skipped in dry run)`);
+            } else {
+                git(`push ${remote} ${branch}`);
+                console.log(`  Created new remote branch: ${branch}`);
+            }
         }
 
         // Next create or update a PR for the branch. Use the remote branch's
         // commit message to create the PR title/body, so that they cannot
         // get out of sync. Note that this will also update the boilerplate of
         // existing PRs even if the branch was not updated. This is intentional.
+        // FIXME: this doesn't work with dry run!
```

Comment:
Problem is that no branches are pushed in dry run, and this uses the remote branches to ensure that they're in sync. Could perhaps change the ref used depending on `dryRun`.